OpenFeature/src/main/java/dev/openfeature/sdk/events/EventPublisher.kt

```diff
@@ -0,0 +1,81 @@
+package dev.openfeature.sdk.events
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.launch
+import kotlin.reflect.KClass
+
+interface ProviderStatus {
+    fun isProviderReady(): Boolean
+}
+
+interface EventObserver {
+    fun <T : OpenFeatureEvents> observe(kClass: KClass<T>): Flow<T>
+}
+
+interface EventsPublisher {
+    fun publish(event: OpenFeatureEvents)
+}
+
+inline fun <reified T : OpenFeatureEvents> EventObserver.observe() = observe(T::class)
+
+class EventHandler(dispatcher: CoroutineDispatcher) : EventObserver, EventsPublisher, ProviderStatus {
+    private val sharedFlow: MutableSharedFlow<OpenFeatureEvents> = MutableSharedFlow()
+    private val isProviderReady = MutableStateFlow(false)
+    private val coroutineScope = CoroutineScope(dispatcher)
+
+    init {
+        coroutineScope.launch {
+            sharedFlow.collect {
+                when (it) {
+                    is OpenFeatureEvents.ProviderReady -> isProviderReady.value = true
+                    is OpenFeatureEvents.ProviderShutDown -> {
```

Comment:
I wonder is the provider is _not_ ready after `ProviderConfigurationChanged` as well

Reply:
like `ShutDownCompleted`?